File: chapter-12-join/code-java/src/main/java/com/example/demo/dto/BookWithAuthor.java
package com.example.demo.dto;

/**
 * 【Chapter 12】 JOIN 결과를 담는 DTO
 * book 테이블과 author 테이블의 컬럼을 하나로 결합합니다.
 */
public record BookWithAuthor(
        Integer id,
        String title,
        Integer publishedYear,
        String firstName,
        String lastName
) {}

File: chapter-12-join/code-kotlin/src/main/kotlin/com/example/demo/dto/BookWithAuthor.kt
package com.example.demo.dto

/**
 * 【Chapter 12】 JOIN 결과를 담는 Kotlin data class DTO
 * fetchInto() 자동 매핑을 위해 null-safe 기본값 포함
 */
data class BookWithAuthor(
    val id: Int? = null,
    val title: String? = null,
    val publishedYear: Int? = null,
    val firstName: String? = null,
    val lastName: String? = null
)
